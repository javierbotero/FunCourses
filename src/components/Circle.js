import React from 'react';
import PropTypes from 'prop-types';

const Circle = props => {
  const {
    quantity,
  } = props;

  const width = 80;
  const height = 80;
  const strWidth = 6;
  const radius = (width / 2) - (strWidth * 2);
  const circumference = radius * 2 * Math.PI;
  const strCircumference = `${circumference} ${circumference}`;
  const completed = circumference - ((circumference / 100) * quantity);

  return (
    <div className="book-child records">
      <svg style={{
        width,
        height,
      }}
      >
        <defs>
          <radialGradient id="gradient" cx="50%" cy="50%" r="70%">
            <stop offset="60%" stopColor="#4333B3" />
            <stop offset="110%" stopColor="#E04534" />
          </radialGradient>
        </defs>
        <circle
          r={radius}
          cx={width / 2}
          cy={height / 2}
          stroke="#e8e8e8"
          style={{
            strokeWidth: strWidth,
            fill: 'transparent',
          }}
        />
        <circle
          r={radius}
          cx={width / 2}
          cy={height / 2}
          stroke="url(#gradient)"
          style={{
            strokeWidth: strWidth,
            strokeDasharray: strCircumference,
            strokeDashoffset: completed,
            fill: 'transparent',
            transform: 'rotate(-90deg)',
            transformOrigin: 'center',
          }}
        />
      </svg>
    </div>
  );
};

Circle.propTypes = {
  quantity: PropTypes.number.isRequired,
};

export default Circle;
