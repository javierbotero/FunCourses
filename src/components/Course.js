import React from 'react';
import { Link } from 'react-router-dom';
import ReactRouterPropTypes from 'react-router-prop-types';
import PropTypes from 'prop-types';

const Course = props => {
  const {
    match,
    courses,
    location,
    url,
  } = props;
  const course = courses.find(c => c.id === parseInt(match.params.id, 10));

  return (
    <div>
      <nav>
        <ul>
          <li>
            <Link to={location.state ? location.state.from.pathname : `${url}`}>
              &#60;
            </Link>
          </li>
        </ul>
      </nav>
      <div>
        <div>{course.title}</div>
        <div>
          <Link
            to={{
              pathname: `${url}/user/${course.teacher_id}`,
              state: { from: location },
            }}
          >
            Teacher:
            {` ${course.teacher.username}`}
          </Link>
        </div>
      </div>
      <div>{course.content}</div>
      <div>
        status:
        {` ${course.status}`}
      </div>
    </div>
  );
};

Course.propTypes = {
  match: ReactRouterPropTypes.match.isRequired,
  location: ReactRouterPropTypes.location.isRequired,
  courses: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
    teacher_id: PropTypes.number.isRequired,
    status: PropTypes.string.isRequired,
    dates: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    created_at: PropTypes.string.isRequired,
    updated_at: PropTypes.string.isRequired,
    teacher: PropTypes.shape({
      id: PropTypes.number,
      username: PropTypes.string,
    }).isRequired,
    favorites: PropTypes.arrayOf(PropTypes.shape({
      course_id: PropTypes.number,
      user_id: PropTypes.number,
    })).isRequired,
    subscriptions: PropTypes.arrayOf(PropTypes.shape({
      course_id: PropTypes.number,
      user_id: PropTypes.number.isRequired,
      confirmed: PropTypes.bool,
    })).isRequired,
    comments: PropTypes.arrayOf(PropTypes.shape({
      user_id: PropTypes.number,
      course_id: PropTypes.number,
      body: PropTypes.string,
    })).isRequired,
    pendings: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number,
      user_id: PropTypes.number,
      course_id: PropTypes.number,
      confirmed: PropTypes.bool,
    })).isRequired,
  })).isRequired,
  url: PropTypes.string.isRequired,
};

export default Course;
