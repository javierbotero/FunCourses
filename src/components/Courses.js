import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import ReactRouterPropTypes from 'react-router-prop-types';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import CoursesCss from '../css/Courses.module.css';
import avatarImage from '../assets/images/avatar.jpg';

const Courses = props => {
  const {
    courses,
    coursesToDivs,
    location,
    url,
    useAuth,
    resetUser,
    resetCourses,
    avatar,
    username,
    urlApi,
    setShowMenu,
  } = props;
  const popUpMenu = () => {
    const div = document.getElementsByClassName('scrollContainer')[0];
    const navMenu = () => document.querySelector('.navMenu');
    if (div.scrollLeft < 200) {
      navMenu().classList.add('hide');
      navMenu().classList.remove('show');
    } else {
      navMenu().classList.remove('hide');
      navMenu().classList.add('show');
    }
  };
  useEffect(() => {
    const div = document.getElementsByClassName('scrollContainer')[0];
    div.addEventListener('scroll', popUpMenu);
    return () => {
      div.removeEventListener('scroll', popUpMenu);
    };
  }, []);
  const authObject = useAuth();
  const logout = () => {
    localStorage.removeItem('currentUserIdFunCourses');
    localStorage.removeItem('currentUserPasswordFunCourses');
    authObject.setUserId(null);
    authObject.setUserPassword(null);
    resetUser();
    resetCourses();
  };

  return (
    <div
      className={`${CoursesCss.courses} scrollContainer`}
    >
      <nav className="navMenu hide">
        <button className="removeButtonStyles" type="button" onClick={() => setShowMenu(true)}>
          <FontAwesomeIcon icon="bars" />
        </button>
      </nav>
      <header className={`${CoursesCss.coursesMenu}`}>
        <div className={`${CoursesCss.avatarZone}`}>
          <div
            className={`avatar ${CoursesCss.avatarPic}`}
            style={{
              backgroundImage: avatar ? `url(${urlApi}${avatar.slice(1, avatar.length)})` : `url(${avatarImage})`,
            }}
          />
          <h4 className={CoursesCss.username}>{username}</h4>
        </div>
        <ul className={`${CoursesCss.menu}`}>
          <li>
            <Link to={{
              pathname: `${url}/dashboard`,
              state: { from: location },
            }}
            >
              Dashboard
            </Link>
          </li>
          <li>
            <Link to={{
              pathname: `${url}/create`,
              state: { from: location },
            }}
            >
              Create
            </Link>
          </li>
          <li>
            <Link to={{
              pathname: `${url}/report`,
              state: { from: location },
            }}
            >
              Report
            </Link>
          </li>
        </ul>
        <button className={`${CoursesCss.btn}`} type="button" onClick={logout}>Log out</button>
      </header>
      <div className={`${CoursesCss.coursesItems}`}>
        {coursesToDivs(courses, url, location)}
      </div>
    </div>
  );
};

Courses.propTypes = {
  courses: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
    teacher_id: PropTypes.number.isRequired,
    status: PropTypes.string.isRequired,
    dates: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    created_at: PropTypes.string.isRequired,
    updated_at: PropTypes.string.isRequired,
    teacher: PropTypes.shape({
      id: PropTypes.number,
      username: PropTypes.string,
    }).isRequired,
    favorites: PropTypes.arrayOf(PropTypes.shape({
      course_id: PropTypes.number,
      user_id: PropTypes.number,
    })).isRequired,
    subscriptions: PropTypes.arrayOf(PropTypes.shape({
      course_id: PropTypes.number,
      user_id: PropTypes.number.isRequired,
      confirmed: PropTypes.bool,
    })).isRequired,
    comments: PropTypes.arrayOf(PropTypes.shape({
      user_id: PropTypes.number,
      course_id: PropTypes.number,
      body: PropTypes.string,
    })),
    pendings: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number,
      user_id: PropTypes.number,
      course_id: PropTypes.number,
      confirmed: PropTypes.bool,
    })).isRequired,
  })).isRequired,
  coursesToDivs: PropTypes.func.isRequired,
  location: ReactRouterPropTypes.location.isRequired,
  url: PropTypes.string.isRequired,
  useAuth: PropTypes.func.isRequired,
  resetUser: PropTypes.func.isRequired,
  resetCourses: PropTypes.func.isRequired,
  avatar: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  urlApi: PropTypes.string.isRequired,
  setShowMenu: PropTypes.func.isRequired,
};

export default Courses;
