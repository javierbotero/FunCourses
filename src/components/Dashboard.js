import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ReactRouterPropTypes from 'react-router-prop-types';
import { Link } from 'react-router-dom';

const Dashboard = props => {
  const {
    user,
    location,
    url,
    coursesToDivs,
    usersListToDiv,
  } = props;

  const displayPendingSubscriptions = (
    course,
    url,
    location,
  ) => course.pendings.map(p => {
    const student = course.pending_students.find(s => s.id === p.user_id);

    return (
      <div className="request" key={p.id}>
        <div>
          {'Course: '}
          <Link to={{
            pathname: `${url}/course/${course.id}`,
            state: { from: location },
          }}
          >
            {course.title}
          </Link>
        </div>
        <div>
          {'Student: '}
          <Link
            to={{
              pathname: `${url}/user/${course.id}`,
              state: { from: location },
            }}
          >
            {student.username}
          </Link>
        </div>
        <div>
          <button type="button">Accept</button>
          <button type="button">Ignore</button>
        </div>
      </div>
    );
  });

  const teacherCoursesToDivs = (courses, url, location) => courses.map(c => (
    <div className="teacher-course" key={c.id}>
      <h4>{c.title}</h4>
      <section>
        <h4>Enrollment Requests</h4>
        {displayPendingSubscriptions(c, url, location)}
        <p>
          {c.pendings.length}
          {' students waiting'}
        </p>
      </section>
      <aside>
        <h4>Accepted Students</h4>
        <div>
          {usersListToDiv(c.confirmed_students)}
        </div>
        <p>
          {c.subscriptions.length}
          {' students are confirmed'}
        </p>
      </aside>
    </div>
  ));

  const displayPendingFriendships = (students, requests, location) => requests.map(r => {
    const student = students.find(s => s.id === r.sender_id);
    return (
      <div className="request" key={r.id}>
        <div style={{ backgroundImage: 'url(#)' }}>
          avatar
        </div>
        <Link
          to={{
            pathname: `${url}/user/${student.id}`,
            state: { from: location },
          }}
        >
          {student.username}
        </Link>
        <div>
          <button type="button">Accept</button>
          <button type="button">Ignore</button>
        </div>
      </div>
    );
  });

  return (
    <div>
      <nav>
        <ul>
          <li>
            <Link to={{
              pathname: location.state ? location.state.from.pathname : `${url}`,
              state: { from: location },
            }}
            >
              &#60;
            </Link>
          </li>
        </ul>
      </nav>
      <header>
        <h2>Dashboard</h2>
        <p>
          Hi
          {` ${user.username} `}
          !
        </p>
      </header>
      <main>
        <h3>Your Courses</h3>
        <div className="student">
          <h4>As a Student</h4>
          {coursesToDivs(user.courses_as_student)}
        </div>
        <div className="teacher">
          <h4>As a Teacher</h4>
          {teacherCoursesToDivs(user.courses, url, location)}
        </div>
      </main>
      <aside>
        <div>
          <h4>
            Friendship Requests
          </h4>
          <div>
            {displayPendingFriendships(
              user.friendship_requests,
              user.pending_to_accept_friendships,
              location,
            )}
          </div>
        </div>
      </aside>
    </div>
  );
};

Dashboard.propTypes = {
  user: PropTypes.shape({
    id: PropTypes.number.isRequired,
    username: PropTypes.string.isRequired,
    courses_as_student: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      content: PropTypes.string.isRequired,
      teacher_id: PropTypes.number.isRequired,
      status: PropTypes.string.isRequired,
      dates: PropTypes.string.isRequired,
      price: PropTypes.number.isRequired,
      created_at: PropTypes.string.isRequired,
      updated_at: PropTypes.string.isRequired,
      teacher: PropTypes.shape({
        id: PropTypes.number,
        username: PropTypes.string,
      }).isRequired,
      favorites: PropTypes.arrayOf(PropTypes.shape({
        course_id: PropTypes.number,
        user_id: PropTypes.number,
      })).isRequired,
      subscriptions: PropTypes.arrayOf(PropTypes.shape({
        course_id: PropTypes.number,
        user_id: PropTypes.number.isRequired,
        confirmed: PropTypes.bool,
      })).isRequired,
      comments: PropTypes.arrayOf(PropTypes.shape({
        user_id: PropTypes.number,
        course_id: PropTypes.number,
        body: PropTypes.string,
      })).isRequired,
      pendings: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.number,
        user_id: PropTypes.number,
        course_id: PropTypes.number,
        confirmed: PropTypes.bool,
      })).isRequired,
    })).isRequired,
    pending_courses_as_student: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      content: PropTypes.string.isRequired,
      teacher_id: PropTypes.number.isRequired,
      dates: PropTypes.string.isRequired,
    })).isRequired,
    comments: PropTypes.arrayOf(PropTypes.shape({
      body: PropTypes.string.isRequired,
      course_id: PropTypes.number.isRequired,
    })).isRequired,
    favorites: PropTypes.arrayOf(PropTypes.shape({
      course_id: PropTypes.number,
    })).isRequired,
    pending_to_accept_friendships: PropTypes.arrayOf(PropTypes.shape({
      receiver_id: PropTypes.number,
      sender_id: PropTypes.number,
    })).isRequired,
    friendship_requests: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number,
      username: PropTypes.string,
      email: PropTypes.string,
    })).isRequired,
    courses: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      content: PropTypes.string.isRequired,
      teacher_id: PropTypes.number.isRequired,
      status: PropTypes.string.isRequired,
      dates: PropTypes.string.isRequired,
      price: PropTypes.number.isRequired,
      created_at: PropTypes.string.isRequired,
      updated_at: PropTypes.string.isRequired,
      teacher: PropTypes.shape({
        id: PropTypes.number,
        username: PropTypes.string,
      }).isRequired,
      favorites: PropTypes.arrayOf(PropTypes.shape({
        course_id: PropTypes.number,
        user_id: PropTypes.number,
      })).isRequired,
      subscriptions: PropTypes.arrayOf(PropTypes.shape({
        course_id: PropTypes.number,
        user_id: PropTypes.number.isRequired,
        confirmed: PropTypes.bool,
      })).isRequired,
      comments: PropTypes.arrayOf(PropTypes.shape({
        user_id: PropTypes.number,
        course_id: PropTypes.number,
        body: PropTypes.string,
      })).isRequired,
      pendings: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.number,
        user_id: PropTypes.number,
        course_id: PropTypes.number,
        confirmed: PropTypes.bool,
      })).isRequired,
    })),
  }).isRequired,
  coursesToDivs: PropTypes.func.isRequired,
  location: ReactRouterPropTypes.location.isRequired,
  url: PropTypes.string.isRequired,
  usersListToDiv: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  user: state.user.user,
});

export default connect(mapStateToProps)(Dashboard);
