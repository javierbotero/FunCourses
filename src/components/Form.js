import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import ReactRouterPropTypes from 'react-router-prop-types';
import {
  setUserError,
  removeUserError,
  setTrueLoading,
  setFalseLoading,
} from '../reducers/user';
import formCss from '../css/Form.module.css';
import fun1 from '../assets/images/fun1.jpeg';
import fun2 from '../assets/images/fun2.jpg';

const Form = props => {
  const {
    initCreator,
    tokenPayload,
    id,
    token,
    url,
    handleApiRequest,
    match,
    setUserErr,
    removeUserErr,
    setTrueLoad,
    setFalseLoad,
    history,
    authenticate,
  } = props;
  const [userName, setUserName] = useState('');
  const [password, setPassword] = useState('');
  const [passConf, setPassConf] = useState('');
  const [email, setEmail] = useState('');
  const [avatar, setAvatar] = useState('');
  const [matcher, setMatcher] = useState(match.params.identifier);
  const handleSubmit = async e => {
    e.preventDefault();
    setTrueLoad();
    if (matcher === 'signup') {
      if (password.length < 5 || userName.length < 5 || email.length < 5 || password !== passConf) {
        setUserErr('Please fill the form correctly, Username and password should be more than 4 characters, password and confirmation must be equal');
        return;
      }
    } else if (password.length < 5 || userName.length < 5) {
      setUserErr('Please fill the form correctly, Username and password should be more than 4 characters');
      return;
    }
    const tokenInfo = tokenPayload(id, token);
    const dataSignUp = {
      ...tokenInfo,
      avatar,
      user: {
        username: userName,
        password,
        password_confirmation: passConf,
        email,
      },
    };
    const dataLogIn = {
      ...tokenInfo,
      username: userName,
      password,
    };
    authenticate(
      handleApiRequest,
      initCreator,
      'POST',
      `${url}${matcher === 'signup' ? 'signup' : 'login'}`,
      matcher === 'signup' ? dataSignUp : dataLogIn,
      setFalseLoad,
    );
    setUserName('');
    setEmail('');
    setPassword('');
    setPassConf('');
  };
  const handleChange = (e, cb, attach = false) => {
    removeUserErr();
    if (attach && e.target.files[0]) {
      const pictureData = new FileReader();
      pictureData.onloadstart = () => setTrueLoad();
      pictureData.onloadend = () => setFalseLoad();
      pictureData.readAsDataURL(e.target.files[0]);
      pictureData.onload = () => {
        cb(
          {
            io: pictureData.result.split(',')[1],
            filename: e.target.files[0].name,
          },
        );
      };
      pictureData.onerror = () => {
        setUserErr('File could not be loaded');
      };
    } else {
      cb(e.target.value);
    }
  };
  const changeHistory = str => {
    setMatcher(str);
    history.push(`/${str}`);
  };

  const infoSign = (signup = false) => {
    if (signup) {
      return (
        <div>
          {'Hello there! join our community and start '}
          {'enjoying fun courses. If you have an account please '}
          <button className="button-3" type="button" onClick={() => changeHistory('login')}>Log in</button>
        </div>
      );
    }
    return (
      <div>
        {'Hello there! it\'s time to learn with fun courses. '}
        {"If you don't have an account please "}
        <button className="button-3" type="button" onClick={() => changeHistory('signup')}>Sign up</button>
      </div>
    );
  };

  return (
    <div
      className="form positionImage"
      style={{ backgroundImage: matcher === 'signup' ? `url(${fun1})` : `url(${fun2})` }}
    >
      <div className={`${formCss.layer} ${matcher === 'signup' ? formCss.signup : formCss.login}`} />
      <h3 className={`${formCss.index2} form-item`}>{matcher === 'signup' ? 'Sign Up' : 'Log in'}</h3>
      <div className={`form-item ${formCss.index2}`}>{matcher === 'signup' ? infoSign(true) : infoSign()}</div>
      <form className={`form-item ${formCss.index2}`} onSubmit={handleSubmit}>
        <div>
          <label htmlFor="username">
            <div>Username</div>
            <input type="text" onChange={e => handleChange(e, setUserName)} value={userName} id="username" placeholder="User" />
          </label>
        </div>
        {matcher === 'signup'
          && (
            <div>
              <label htmlFor="email">
                <div>Email</div>
                <input type="email" onChange={e => handleChange(e, setEmail)} value={email} id="email" placeholder="user@mail.com" />
              </label>
            </div>
          )}
        <div>
          <label htmlFor="password">
            <div>Password</div>
            <input type="password" onChange={e => handleChange(e, setPassword)} value={password} id="password" placeholder="password" />
          </label>
        </div>
        { matcher === 'signup' && (
          <div>
            <label htmlFor="passwordConfirmation">
              <div>Password confirmation</div>
              <input type="password" onChange={e => handleChange(e, setPassConf)} value={passConf} id="passwordConfirmation" placeholder="confirmation" />
            </label>
          </div>
        )}
        { matcher === 'signup' && (
          <div>
            <label htmlFor="avatar">
              <div>Image Profile</div>
              <input type="file" alt="avatar" onChange={e => handleChange(e, setAvatar, true)} id="avatar" name="avatar" accept="image/*" />
            </label>
          </div>
        )}
        <div className={`${formCss.buttonSubmit}`}>
          <input type="submit" value="Submit" className="btn button-2" />
        </div>
      </form>
    </div>
  );
};

Form.propTypes = {
  initCreator: PropTypes.func.isRequired,
  tokenPayload: PropTypes.func.isRequired,
  url: PropTypes.string.isRequired,
  token: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  handleApiRequest: PropTypes.func.isRequired,
  match: ReactRouterPropTypes.match.isRequired,
  setUserErr: PropTypes.func.isRequired,
  removeUserErr: PropTypes.func.isRequired,
  setTrueLoad: PropTypes.func.isRequired,
  setFalseLoad: PropTypes.func.isRequired,
  history: ReactRouterPropTypes.history.isRequired,
  authenticate: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  setUserErr: err => dispatch(setUserError(err)),
  removeUserErr: () => dispatch(removeUserError()),
  setTrueLoad: () => dispatch(setTrueLoading()),
  setFalseLoad: () => dispatch(setFalseLoading()),
});

export default connect(null, mapDispatchToProps)(Form);
export { Form };
