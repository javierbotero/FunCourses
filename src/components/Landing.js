import React from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { OPINIONS } from '../constants/constants';
import image1 from '../assets/images/fun1.jpeg';
import styles from '../css/Landing.module.css';

const Landing = () => {
  const displayOpinions = opinions => opinions.map(o => (
    <div key={o.avatar} className="opinion">
      <div style={{
        backgroundImage: o.avatar,
      }}
      />
      <div>{o.body}</div>
    </div>
  ));

  return (
    <div className="info-landing">
      <header className={`${styles.containers} ${styles.containerHeader}`} style={{ backgroundImage: `url(${image1})` }}>
        <nav className={`${styles.navHeader}`}>
          <div className="logo">Logo</div>
          <ul className={`${styles.loginMenu}`}>
            <li><Link to="/signup">Sign up</Link></li>
            <li><Link to="/login">Log in</Link></li>
          </ul>
        </nav>
        <h4 className={styles.headerPar}>Find a course of any theme</h4>
        <p>
          The best courses for you about any
          theme you can think of, taught by
          professionals.
        </p>
        <Link to="/signin">Start Free</Link>
      </header>
      <main>
        <section className={`${styles.containers} ${styles.verticalCenter} bg-white`}>
          <h4>Interesting courses from reliable teachers</h4>
          <div>
            <div>
              <h4>
                All stats at your fingerprints
              </h4>
              <p>
                All the information about the number
                of students, your courses, comments
                and interactions will be available to
                you to measure your performance.
              </p>
            </div>
            <div className="stats" />
          </div>
        </section>
        <section className={`${styles.containers} ${styles.verticalCenter} bg-purple`}>
          <div className="stats" />
          <div>
            <div>
              <h4>
                Talk with professionals
              </h4>
              <p>
                The best professionals around the world will
                be reachable through live courses. take the
                opportunity to enhance your skills in any matter.
              </p>
            </div>
          </div>
        </section>
        <section className={`${styles.containers} ${styles.verticalCenter} bg-white`}>
          <div>PEOPLE ARE ALREADY TAKING FUN COURSES</div>
          <h4>Look what these are saying</h4>
          <div className="opinions">
            {displayOpinions(OPINIONS)}
          </div>
        </section>
      </main>
      <footer className={`${styles.footerLook} bg-white`}>
        <div className={`${styles.socialImage}`} style={{ backgroundImage: `url(${image1})` }}>
          <h4>Learn everything at home</h4>
          <Link to="/signin">Start free</Link>
        </div>
        <div className={`${styles.verticalCenter} ${styles.socialMedia}`}>
          <div>
            <div className="logo" />
            <div>Fun Courses</div>
          </div>
          <div>
            <FontAwesomeIcon icon={['fab', 'linkedin']} />
            <FontAwesomeIcon icon={['fab', 'facebook']} />
            <FontAwesomeIcon icon={['fab', 'twitter-square']} />
            <FontAwesomeIcon icon={['fab', 'instagram-square']} />
          </div>
        </div>
      </footer>
    </div>
  );
};

export default Landing;
