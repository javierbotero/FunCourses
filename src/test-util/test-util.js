import fetchMock from 'fetch-mock';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { URL } from '../constants/constants';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const mockMatch = {
  params: { id: '5' },
  path: 'some-path',
  url: 'url',
};
const mockCourses = [
  {
    id: 5,
    title: 'Karate class',
    content: 'Nice Karate class',
    teacher_id: 7,
    status: 'Open',
    dates: '2021-03-25T00:00:00.000Z 2021-03-29T00:00:00.000Z ',
    price: 10,
    images_url: [],
    created_at: '2021-03-07T17:56:22.487Z',
    updated_at: '2021-03-15T04:21:24.489Z',
    teacher: {
      id: 7,
      username: 'Adele',
    },
    pendings: [],
    pending_students: [],
    favorites: [],
    subscriptions: [],
    confirmed_students: [],
  },
];
const mockHistory = {
  action: 'PUSH',
  block: jest.fn(),
  createHref: jest.fn(),
  go: jest.fn(),
  listen: jest.fn(),
  push: jest.fn(),
  location: {
    hash: '',
    pathname: '',
    search: '',
  },
  replace: jest.fn(),
};
const mockUser = {
  id: 7,
  status: 'Student',
  username: 'Javier 4',
  url_avatar: 'api.com/users/1',
  courses_as_student: [],
  pending_courses_as_student: [],
  comments: [],
  favorites: [],
  pending_to_accept_friendships: [],
  pending_requested_friendships: [],
  friendship_requests: [],
  requests: [],
  pendings: [],
  courses: [
    {
      id: 5,
      title: 'Karate class',
      content: 'Nice Karate class',
      teacher_id: 7,
      status: 'Open',
      dates: '2021-03-25T00:00:00.000Z 2021-03-29T00:00:00.000Z ',
      price: 10,
      created_at: '2021-03-07T17:56:22.487Z',
      updated_at: '2021-03-15T04:21:24.489Z',
      teacher: {
        id: 7,
        username: 'Adele',
      },
      pendings: [],
      pending_students: [],
      favorites: [],
      subscriptions: [],
      confirmed_students: [],
    },
  ],
};
const mockTeacherCourses = [];
const mockLocation = {
  from: { pathname: 'url' },
  hash: 'some-hash',
  pathname: 'url',
  search: 'default',
  state: {
    user: {
      id: 1,
      username: 'Paco',
    },
    from: {
      pathname: 'url',
    },
  },
};
const mockUrl = 'url';
const mockUsersListToDiv = jest.fn();
const mockCommentsToDivs = jest.fn();
const userPassword = '12345';
const func = jest.fn();
const mockUrlApi = 'url';
const mockToken = 'token';
const mockId = 1;

export {
  func,
  mockStore,
  fetchMock,
  mockMatch,
  mockCourses,
  mockTeacherCourses,
  mockLocation,
  mockUrl,
  mockUsersListToDiv,
  mockCommentsToDivs,
  mockUrlApi,
  mockToken,
  mockId,
  mockUser,
  URL,
  mockHistory,
  userPassword,
};
