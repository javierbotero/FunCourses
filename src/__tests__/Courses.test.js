import React from 'react';
import {
  render,
  fireEvent,
  screen,
} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import * as util from '../test-util/test-util';
import { Courses } from '../components/Courses';

describe('Tests for courses', () => {
  it('Makes sure Dasboard text is present in the document', () => {
    render(
      <BrowserRouter>
        <div className="desktopMenu" />
        <Courses
          urlApi="urlApi"
          setShowMenu={util.func}
          username="Javier"
          avatar="avatar"
          courses={util.mockCourses}
          location={util.mockLocation}
          url={util.mockUrl}
          coursesToDivs={util.func}
          resetUser={util.func}
          resetCourses={util.func}
        />
      </BrowserRouter>,
    );
    expect(screen.getByText('Dashboard')).toBeInTheDocument();
  });
  it('Makes sure coursesToDivs method is called when render Courses', () => {
    render(
      <BrowserRouter>
        <div className="desktopMenu" />
        <Courses
          urlApi="urlApi"
          setShowMenu={util.func}
          username="Javier"
          avatar="avatar"
          courses={util.mockCourses}
          location={util.mockLocation}
          url={util.mockUrl}
          coursesToDivs={util.func}
          resetUser={util.func}
          resetCourses={util.func}
        />
      </BrowserRouter>,
    );
    expect(util.func.mock.calls.length).toBe(1);
  });
  it('Renders Courses component', () => {
    render(
      <BrowserRouter>
        <div className="desktopMenu" />
        <Courses
          urlApi="urlApi"
          setShowMenu={util.func}
          username="Javier"
          avatar="avatar"
          courses={util.mockCourses}
          location={util.mockLocation}
          url={util.mockUrl}
          coursesToDivs={util.func}
          resetUser={util.func}
          resetCourses={util.func}
        />
      </BrowserRouter>,
    );
    expect(util.func.mock.calls.length).toBe(1);
  });
  it('Renders Courses component', () => {
    const myFunc = jest.fn();
    render(
      <BrowserRouter>
        <div className="desktopMenu" />
        <Courses
          urlApi="urlApi"
          setShowMenu={util.func}
          username="Javier"
          avatar="avatar"
          courses={util.mockCourses}
          location={util.mockLocation}
          url={util.mockUrl}
          coursesToDivs={util.func}
          resetUser={myFunc}
          resetCourses={util.func}
        />
      </BrowserRouter>,
    );
    fireEvent.click(screen.getByRole('button', { name: /Log out/i }));
    expect(myFunc.mock.calls.length).toBe(1);
  });
  it('Renders Courses component', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <div className="desktopMenu" />
          <Courses
            urlApi="urlApi"
            setShowMenu={util.func}
            username="Javier"
            avatar="avatar"
            courses={util.mockCourses}
            location={util.mockLocation}
            url={util.mockUrl}
            coursesToDivs={util.func}
            resetUser={util.func}
            resetCourses={util.func}
          />
        </BrowserRouter>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
