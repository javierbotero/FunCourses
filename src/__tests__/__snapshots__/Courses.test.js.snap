// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for courses Renders Courses component 1`] = `
Array [
  <div
    className="desktopMenu"
  />,
  <div
    className="courses scrollContainer"
  >
    <nav
      className="navMenu hide"
    >
      <button
        className="removeButtonStyles"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa bars"
        />
      </button>
    </nav>
    <header
      className="coursesMenu"
    >
      <div
        className="avatarZone"
      >
        <div
          className="avatar avatarPic"
          style={
            Object {
              "backgroundImage": "url(urlApivatar)",
            }
          }
        />
        <h4
          className="username"
        >
          Javier
        </h4>
      </div>
      <ul
        className="menu"
      >
        <li>
          <a
            href="url/dashboard"
            onClick={[Function]}
          >
            Dashboard
          </a>
        </li>
        <li>
          <a
            href="url/create"
            onClick={[Function]}
          >
            Create
          </a>
        </li>
        <li>
          <a
            href="url/report"
            onClick={[Function]}
          >
            Report
          </a>
        </li>
      </ul>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        Log out
      </button>
    </header>
    <div
      className="coursesItems"
    />
  </div>,
]
`;
